### DISCUSSION TITLE
good function/way to put diagnostic on screen without corrupting it?

### DISCUSSION THREAD

**Discussion by bkerin** [akinomyoga/ble.sh#306] [General]
I've been making bits of progress on my blesh setup and a lot of it Seems To Be Working (tm) but now I'd like to leave some traps at various points where I'm still uncertain.  Right now I'm using this:

     function brokenFail { echo -e "\n$@\n" 1>&2; ble/widget/redraw-line; return 1; }

but it's obviously quite bad and destroys command line in progress.  I'd like to have a trap that writes something obvious to the screen somehow preferably without demolishing the command I'm working on at the time.  Is there already something like this somewhere, or an easy way to make it?  I don't know if it would be possible for it to unwind to the state before the last widget or something but would be nice and maybe that's how undo works anyway or something.

**Comment by akinomyoga** [comment]
I'm not sure if I correctly understand what you want to achieve, but does the following work as you expect?

```bash
function brokenFail { ble/widget/print "$*"; return 1; }
```

**Edit**: Or, if you would like to show the visible bell, you can write it as follows (where `bleopt edit_vbell=1` needs to be turned on (default)):

```bash
function brokenFail { ble/widget/bell "$*"; return 1; }
```

Another way is to show the message in the *info* panel below the command line:

```bash
function brokenFail { ble/edit/info/show text "$*"; return 1; }
```

The *info* panel is the one where `-- INSERT --` indicators, history search progress, menu completions, etc. are shown. By `ble/edit/info/show`, you can temporarily change the content of the *info* panel.

With the above, the message will be shown in the next cycle of the rendering. If you would like to immediately show the message, you can use the following instead.

```bash
function brokenFail { ble/edit/info/immediate-show text "$*"; return 1; }
```

**Comment by bkerin** [comment]
yes that's much better thx
